packages/devtools_app/lib/src/flutter/split.dart

```diff
@@ -100,29 +101,70 @@ class _SplitState extends State<Split> {
     final height = constraints.maxHeight;
     final axisSize = isHorizontal ? width : height;
     final crossAxisSize = isHorizontal ? height : width;
-    const halfDivider = Split.dividerMainAxisSize / 2.0;
 
     // Determine what fraction to give each child, including enough space to
     // display the divider.
-    double firstSize = axisSize * firstFraction;
-    double secondSize = axisSize * secondFraction;
-
-    // Clamp the sizes to be sure there is enough space for the dividers.
-    firstSize = firstSize.clamp(halfDivider, axisSize - halfDivider);
-    secondSize = secondSize.clamp(halfDivider, axisSize - halfDivider);
+    final numDividers = widget.children.length - 1;
+    final sizes = List.generate(
+      fractions.length,
+      (i) =>
+          (axisSize - numDividers * Split.dividerMainAxisSize) * fractions[i],
+    );
 
-    // Remove space from each child to place the divider in the middle.
-    firstSize = firstSize - halfDivider;
-    secondSize = secondSize - halfDivider;
+    void updateSpacing(DragUpdateDetails dragDetails, int splitterIndex) {
+      final dragDelta =
+          isHorizontal ? dragDetails.delta.dx : dragDetails.delta.dy;
+      final fractionalDelta = dragDelta / axisSize;
+
+      double updateSpacingBeforeSplitterIndex(double delta) {
```

Comment:
comment that these two methods return the actual delta applied.

Reply:
> comment that these two methods return the actual delta applied.

I misread what the method did. I mean change the implementation to return that as that is simpler than returning the delta remaining which makes the logic a bit confusing.

Reply:
good idea. done.